internal/provider/resource_tfe_team_notification_configuration.go

```diff
@@ -0,0 +1,437 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-provider-tfe/internal/provider/validators"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &resourceTFETeamNotificationConfiguration{}
+var _ resource.ResourceWithConfigure = &resourceTFETeamNotificationConfiguration{}
+var _ resource.ResourceWithImportState = &resourceTFETeamNotificationConfiguration{}
+
+func NewTeamNotificationConfigurationResource() resource.Resource {
+	return &resourceTFETeamNotificationConfiguration{}
+}
+
+// resourceTFETeamNotificationConfiguration implements the tfe_team_notification_configuration resource type
+type resourceTFETeamNotificationConfiguration struct {
+	config ConfiguredClient
+}
+
+func (r *resourceTFETeamNotificationConfiguration) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team_notification_configuration"
+}
+
+type modelTFETeamNotificationConfiguration struct {
+	ID              types.String `tfsdk:"id"`
+	Name            types.String `tfsdk:"name"`
+	DestinationType types.String `tfsdk:"destination_type"`
+	EmailAddresses  types.Set    `tfsdk:"email_addresses"`
+	EmailUserIDs    types.Set    `tfsdk:"email_user_ids"`
+	Enabled         types.Bool   `tfsdk:"enabled"`
+	Token           types.String `tfsdk:"token"`
+	Triggers        types.Set    `tfsdk:"triggers"`
+	URL             types.String `tfsdk:"url"`
+	TeamID          types.String `tfsdk:"team_id"`
+}
+
+// modelFromTFETeamNotificationConfiguration builds a modelTFETeamNotificationConfiguration
+// struct from a tfe.TeamNotificationConfiguration value.
+func modelFromTFETeamNotificationConfiguration(v *tfe.NotificationConfiguration) modelTFETeamNotificationConfiguration {
+	result := modelTFETeamNotificationConfiguration{
+		ID:              types.StringValue(v.ID),
+		Name:            types.StringValue(v.Name),
+		DestinationType: types.StringValue(string(v.DestinationType)),
+		Enabled:         types.BoolValue(v.Enabled),
+		TeamID:          types.StringValue(v.SubscribableChoice.Team.ID),
+	}
+
+	if emailAddresses, err := types.SetValueFrom(ctx, types.StringType, v.EmailAddresses); err == nil {
+		result.EmailAddresses = emailAddresses
+	}
+
+	if len(v.Triggers) == 0 {
+		result.Triggers = types.SetNull(types.StringType)
+	} else if triggers, err := types.SetValueFrom(ctx, types.StringType, v.Triggers); err == nil {
+		result.Triggers = triggers
+	}
```

Comment:
In both cases here, is the `err != nil` case worth a diagnostic warning? Or are we sure that won't happen based on how the config is parsed in the methods below?

Reply:
I ended up just exposing diagnostics to the caller generating the model and halting if there are any errors